### ISSUE TITLE
[JENKINS-71173] Tables are not initialized correctly

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/data-tables-api-plugin#663]
<p>Opening <a href="https://jenkins-fsn1.samknows.com/job/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins-fsn1.samknows.com/job/</a><job>/coverage/ (from Code Coverage API Plugin) and clicking on "File Coverage".</p>
<ul>
	<li>Works the first time</li>
	<li>Results in an empty file list the following times</li>
</ul>

<p>This is not fixed by restarting the browser or Jenkins. But,</p>
<ul>
	<li>Opening a "New private window" in Firefox allow me to see the file list again (once). I can do this as many times as I want.</li>
	<li>Opening the page with a different browser also allows me to see the file list, also only once.</li>
</ul>

<p>I know zero Javascript / web development. But I have seen</p>

```
jQuery.Deferred exception: cannot use 'in' operator to search for "length" in "[[4, \"desc\"]]" p@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:1092
each@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:2965
pe@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:45761
he@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:45140
w/<@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:8666
each@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:3003
each@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:1481
w@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:3568
P.fn.DataTable@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:86702
createDataTable@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:70:37
bindTables/<@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:109:46
each@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:3003
each@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:1481
bindTables@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:106:19
@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:205:19
@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:206:7
e@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:30158
Deferred/then/l/</t<@​https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:30460
 undefined
```
<p>and</p>

```
Uncaught TypeError: cannot use 'in' operator to search for "length" in "[[4, \"desc\"]]"
    jQuery 9
        p
        each
        pe
        he
        w
        each
        each
        w
        DataTable
    createDataTable https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:70
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:109
    jQuery 2
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:106
    <anonymous> https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:205
    <anonymous> https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:206
    jQuery 2
        e
        t
jquery.min.js:2:1092
    jQuery 9
    createDataTable https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:70
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:109
    jQuery 2
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:106
    <anonymous> https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:205
    <anonymous> https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:206
    jQuery 2
        e
        t
 
```
<p>in the Firefox "Browser Console" (see the original in the attached jenkins_issue.png) which, from the chat in Gitter, apparently means "Somehow the tables are not initialized correctly".</p>

<p> </p>

<p>As of this writing, I'm using the latest versions of all the plugins</p>
<ul>
	<li>Jenkins 2.387.2</li>
	<li>Code Coverage API Plugin Version 4.4.0</li>
	<li>jQuery plugin Version 1.12.4-1</li>
	<li>JQuery3 API Plugin Version 3.6.4-1</li>
	<li>Cobertura Plugin Version 1.17</li>
	<li><a href="http://datatables.net/" class="external-link" target="_blank" rel="nofollow noopener">DataTables.net</a> API Plugin Version 1.13.3-3</li>
</ul>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=reddwarf94">reddwarf94</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-71173" target="_blank">Tables are not initialized correctly</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli">drulli</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: data-tables-api-plugin
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2023-09-21T06:26:20+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 20260107-220949
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Opening <a href="https://jenkins-fsn1.samknows.com/job/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins-fsn1.samknows.com/job/</a>&lt;job&gt;/coverage/ (from Code Coverage API Plugin) and clicking on "File Coverage".</p>
<ul>
	<li>Works the first time</li>
	<li>Results in an empty file list the following times</li>
</ul>

<p>This is not fixed by restarting the browser or Jenkins. But,</p>
<ul>
	<li>Opening a "New private window" in Firefox allow me to see the file list again (once). I can do this as many times as I want.</li>
	<li>Opening the page with a different browser also allows me to see the file list, also only once.</li>
</ul>

<p>I know zero Javascript / web development. But I have seen</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>jQuery.Deferred exception: cannot use 'in' operator to search for "length" in "[[4, \"desc\"]]" p@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:1092
each@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:2965
pe@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:45761
he@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:45140
w/&lt;@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:8666
each@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:3003
each@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:1481
w@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:3568
P.fn.DataTable@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/jquery.dataTables.min.js:4:86702
createDataTable@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:70:37
bindTables/&lt;@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:109:46
each@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:3003
each@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:1481
bindTables@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:106:19
@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:205:19
@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:206:7
e@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:30158
Deferred/then/l/&lt;/t&lt;@https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/jquery3-api/webjars/jquery.min.js:2:30460
 undefined</pre>
</div></div>
<p>and</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Uncaught TypeError: cannot use 'in' operator to search for "length" in "[[4, \"desc\"]]"
    jQuery 9
        p
        each
        pe
        he
        w
        each
        each
        w
        DataTable
    createDataTable https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:70
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:109
    jQuery 2
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:106
    &lt;anonymous&gt; https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:205
    &lt;anonymous&gt; https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:206
    jQuery 2
        e
        t
jquery.min.js:2:1092
    jQuery 9
    createDataTable https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:70
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:109
    jQuery 2
    bindTables https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:106
    &lt;anonymous&gt; https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:205
    &lt;anonymous&gt; https://jenkins-fsn1.samknows.com/static/277dfae1/plugin/data-tables-api/js/table.js:206
    jQuery 2
        e
        t
 </pre>
</div></div>
<p>in the Firefox "Browser Console" (see the original in the attached jenkins_issue.png) which, from the chat in Gitter, apparently means "Somehow the tables are not initialized correctly".</p>

<p> </p>

<p>As of this writing, I'm using the latest versions of all the plugins</p>
<ul>
	<li>Jenkins 2.387.2</li>
	<li>Code Coverage API Plugin Version 4.4.0</li>
	<li>jQuery plugin Version 1.12.4-1</li>
	<li>JQuery3 API Plugin Version 3.6.4-1</li>
	<li>Cobertura Plugin Version 1.17</li>
	<li><a href="http://datatables.net/" class="external-link" target="_blank" rel="nofollow noopener">DataTables.net</a> API Plugin Version 1.13.3-3</li>
</ul>
</pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [jenkins_issue.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/60346/jenkins_issue.png)
  > ![jenkins_issue.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/60346/jenkins_issue.png)
</details>
